urllib3/contrib/socks.py

```diff
@@ -152,6 +152,10 @@ def __init__(self, proxy_url, username=None, password=None,
                  num_pools=10, headers=None, **connection_pool_kw):
         parsed = parse_url(proxy_url)
 
+        if username is None and password is None and parsed.auth is not None:
+            split = parsed.auth.split(':')
+            if len(split) == 2:
```

Comment:
If it's not two parts is it acceptable for us to silently ignore this or should we raise a `ValueError`?

Reply:
I guess (if this question for me) it is not neccesary, because in this case we will have name or password is None, so it will work like as before. In my opinion, correctness of auth info should be in parse_url, probably parse_url should return not auth but pair of username/password, I am not sure. Never mind, if we need ValueError for wrong auth data - it should be in parse_url, not here

Reply:
Sounds good.